### PR TITLE
Feature/better event assertions

### LINKED ISSUE #74: assertions for claim and optionIds
testEventNewOptionType should assert first topic

testEventWrite should assert claim id

testEvent_sweepFees_whenFeesAccruedForExercise should be written

### PR DESCRIPTION
**Pull request by neodaoist**
Resolves #74 

- One production code change — Include claimId as a topic on OptionsWritten event (it's indexed on other events too, so thought this makes sense)
- Fix broken first topic assertion on event test for creating new option type
- Fix claim ID assertion on event emission when writing both new and existing claims
- Add event test for fees swept during exercise

The snapshot diff is way nastier than my changes, think it includes some other recent changes, just heads up. Indexing claimId on the event should only cost 110 gas more on write() calls.

### CODE REVIEW

File: src/test/OptionSettlement.t.sol [hunk 1]
```diff
@@ -470,38 +470,65 @@ contract OptionSettlementTest is Test, NFTreceiver {
...
+        uint256 expectedFeeAccruedAmount = ((testUnderlyingAmount / 10_000) * engine.feeBps());
+
+        vm.prank(ALICE);
+        uint256 claimId = engine.write(testOptionId, 1);
+
+        vm.expectEmit(true, true, true, true);
+        emit FeeAccrued(WETH_A, ALICE, expectedFeeAccruedAmount);
+
+        vm.expectEmit(true, true, true, true);
+        emit OptionsWritten(testOptionId, ALICE, claimId, 1);
```
Comment (Flip-Liquid) [inline src/test/OptionSettlement.t.sol:526]: @0xAlcibiades @neodaoist what do we think makes the most sense here to emit. Currently the index of the claim is emitted (1, 2 ,3 ,4). Do we want to emit the entire claim ID instead with the encoded option ID in the top 160b?

Comment (Flip-Liquid) [reply src/test/OptionSettlement.t.sol:526]: (Not related to changes in this PR btw)

File: src/test/OptionSettlement.t.sol [hunk 2]
```diff
@@ -543,11 +570,16 @@ contract OptionSettlementTest is Test, NFTreceiver {
             uint96(expectedUnderlyingAmount)
             );
 
-        engine.claim(claimId);
+        // engine.claim(claimId);
```
Comment (Flip-Liquid) [inline src/test/OptionSettlement.t.sol:573]: Was the test passing before this? Should we add or update an assert here?

File: src/test/OptionSettlement.t.sol [hunk 3]
```diff
@@ -593,6 +621,88 @@ contract OptionSettlementTest is Test, NFTreceiver {
             emit FeeSwept(tokens[i], engine.feeTo(), expectedFees[i] - 1); // sweeps 1 wei less as gas optimization
         }
 
+        // When fees are swept
+        engine.sweepFees(tokens);
```
Comment (Flip-Liquid) [inline src/test/OptionSettlement.t.sol:625]: Thanks for adding this. Surprised the test wasn't failing before

Comment (Flip-Liquid) [inline src/test/OptionSettlement.t.sol:625]: Thanks for adding this. Surprised the test wasn't failing before

Comment (Flip-Liquid) [reply src/test/OptionSettlement.t.sol:625]: Nvm, git diff hid sweep call

**Comment by 0xAlcibiades** [comment]
Yeah, I’d think so.

File: src/test/OptionSettlement.t.sol (hunk 1 above)
Comment (neodaoist) [reply src/test/OptionSettlement.t.sol:526]: Honestly I would love to talk through how dapp devs or end users might want to view and monitor these events. Need more context to have an opinion yet

File: src/test/OptionSettlement.t.sol (hunk 2 above)
Comment (neodaoist) [reply src/test/OptionSettlement.t.sol:573]: Ah woops, I meant to remove this altogether. There's no need to call the view function, the test exercises the events emitted from redeem().

File: src/test/OptionSettlement.t.sol (hunk 3 above)
Comment (neodaoist) [reply src/test/OptionSettlement.t.sol:625]: Think I just added the comment. This test was calling sweepFees() 👍

Comment (neodaoist) [reply src/test/OptionSettlement.t.sol:625]: Ah see your other comment. Yeah just some diff weirdness

**Review by 0xAlcibiades — APPROVED** [review]

**Pull request merged**